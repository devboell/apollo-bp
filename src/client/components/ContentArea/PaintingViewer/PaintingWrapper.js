import styled from 'styled-components'

const PaintingWrapper = styled.div`
  display: flex;
  justify-content: center;
`

export default PaintingWrapper
