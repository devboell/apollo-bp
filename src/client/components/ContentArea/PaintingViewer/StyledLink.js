import styled from 'styled-components'
import { Link } from 'react-router-dom'

const Styledlink = styled(Link)`
  display: flex;
  text-decoration: none;
  align-items: center;
  height: 40px;
  margin-bottom: 20px;
  border: 1px solid black;
  border-radius: 5px;

  > p {
    color: black;
    font-weight: bold;
    margin-left: 5px;
  }
`

export default Styledlink
