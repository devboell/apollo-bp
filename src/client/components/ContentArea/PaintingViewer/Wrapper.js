import styled from 'styled-components'

const Wrapper = styled.div.attrs({
  'data-testid': 'paintingViewer',
})`
  display: flex;
  flex-flow: column;
  justify-content: flex-start;
  width: 100%;
  margin: auto;
`

export default Wrapper
