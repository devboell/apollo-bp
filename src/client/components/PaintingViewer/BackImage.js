import styled from 'styled-components'

const BackImage = styled.img`
  margin-left: 10px;
  height: 20px;
`

export default BackImage
