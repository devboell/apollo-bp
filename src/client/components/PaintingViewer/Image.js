import styled from 'styled-components'

const Image = styled.img`
  max-width: 100%;
  max-height: 700px;
  margin: auto;
`

export default Image
