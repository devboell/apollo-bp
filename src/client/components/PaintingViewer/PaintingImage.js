import styled from 'styled-components'

const Image = styled.img`
  width: 100%;
  max-height: 500px;
  object-fit: contain;
`

export default Image
