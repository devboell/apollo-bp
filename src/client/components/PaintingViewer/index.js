import React from 'react'
import PropTypes from 'prop-types'

import PaintingImage from './PaintingImage'
import PaintingWrapper from './PaintingWrapper'
import BackImage from './BackImage'
import Wrapper from './Wrapper'
import StyledLink from './StyledLink'
import backImage from './images/back.png'

const PaintingViewer = ({ match }) =>
  <Wrapper>
    <StyledLink to={`/${match.params.artistPath}`}>
      <BackImage src={`../${backImage}`} alt="" />
      <p>back to gallery</p>
    </StyledLink>
    <PaintingWrapper>
      <PaintingImage src={`/images/artists${match.url}`} alt="" />
    </PaintingWrapper>
  </Wrapper>

PaintingViewer.propTypes = {
  match: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
}

export default PaintingViewer
