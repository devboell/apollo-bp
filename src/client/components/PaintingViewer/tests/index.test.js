import React from 'react'
import PaintingViewer from '../index'

const props = {
  match: {
    url: 'testUrl',
  },
}

it('PaintingViewer component snapshot', () => {
  const wrapper = shallow(
    <PaintingViewer {...props} />,
  )
  expect(wrapper).toMatchSnapshot()
})
