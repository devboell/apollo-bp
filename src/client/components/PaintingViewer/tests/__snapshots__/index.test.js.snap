// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaintingViewer component snapshot 1`] = `
<styled.img
  alt=""
  src="/images/artiststestUrl"
/>
`;
